hazelcast-jet-core/src/main/java/com/hazelcast/jet/pipeline/file/JsonFileFormat.java

```diff
@@ -55,6 +56,28 @@
         return this;
     }
 
+    /**
+     * Specifies if the Json parser should accept json records spanning
+     * multiple lines.
+     * <p>
+     * The parser handles JSON records spanning multiple lines by default,
+     * but it prevents reading the file in parallel when using the Hadoop
```

Comment:
I think so. 

You are right that the split happens before the creation of the record reader, but it happens at arbitrary position

Let's say you have the following jsonl file:
```
{
  "name":"Frantisek", 
  "favoriteNumber": 7
}
<here>
{
 <here2>
  "name":"Ali",
  "favoriteNumber": 42
}
```
the split could happen at the place I marked with <here>, but that is optimistic scenario, it could have been at  <here2>, or even in the middle of the field.

So the record reader receiving the first split is required to seek after the end of the split to read full record and the record reader receiving the second split is required to seek to the next record., with json like this the only way to correctly identify when newline means new record it would need to read the whole file to know the exact level of nesting. Therefore to read use parallel reading we just forbid multiline json, then it is safe to use newline as record delimiter. This is what LineRecordReader does. the SingleLineJsonRecordReader delegates to it and just adds json parsing.